crawler/crawlutils.py

```diff
@@ -539,19 +569,5 @@ def snapshot(
         if frequency < 0 or should_exit:
             logger.info('Bye')
             break
-        elif frequency == 0:
-            continue
-
-        if next_iteration_time is None:
-            next_iteration_time = snapshot_time + frequency
-        else:
-            next_iteration_time = next_iteration_time + frequency
-
-        while next_iteration_time + frequency < time.time():
-            next_iteration_time = next_iteration_time + frequency
-
-        time_to_sleep = next_iteration_time - time.time()
-        if time_to_sleep > 0:
-            time.sleep(time_to_sleep)
```

Comment:
This logic is definitely needed. The idea is that if we set a frequency of 5 seconds and it took us 2 seconds to run an iteration, the next sleep should be of 3 seconds. This logic takes care of that.